GibbsSampling.Library/CompositeVector/CompositeVectorTypes.fs

```diff
@@ -0,0 +1,73 @@
+﻿namespace CompositeVector
+
+open System
+open BioFSharp
+
+module Types =
+    
+    /// Matrix with fixed positions for nucleotides and amino acids.
+    type BaseMatrix<'a, 'value when 'a :> IBioItem>internal(matrix:'value [,]) =
```

Comment:
Here you are always working on the original matrix given into the BaseMatrix type.

It might be beneficial store a copy of it inside the type to increase robustness against mutable changes. But this depends on the performance of course.

```fsharp
type BaseMatrix (matrix) =
    member val _matrix = matrix.Copy() with get, set
```

Here copy represents any logic necessary to copy the mutable object. then work on `_matrix` instead of `matrix`

Reply:
fixed